pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dao/JdbiBlockTabular.java

```diff
@@ -58,18 +56,27 @@ long insertQuestion(@Bind("tabularBlockId") long tabularBlockId,
             + "  JOIN revision as rev ON rev.revision_id = bt.revision_id"
             + " WHERE bt.block_id = :blockId"
             + "   AND rev.start_date <= :timestamp"
-            + "   AND (rev.end_date is null or :timestamp < rev.end_date)")
+            + "   AND (rev.end_date is null or :timestamp < rev.end_date)"
+            + " order by bth.block_tabular_header_id")
```

Comment:
Let's not change this now, but there is an assumption we are leaning on here that keys are incremental and that insertions will happen in order. A dedicated sort order key/column is the way to go.

(Was looking how we had done it before for conditional block children, and this kind of funny: the blocks have sort key, but the section does not so we are using the primary key!)

Again, let's not mess with this now. We can take another look at some point.

https://github.com/broadinstitute/ddp-study-server/blob/355b603ee1eac09d6c0c3253f590c8b58cc11b2e/pepper-apis/dss-core/src/main/resources/org/broadinstitute/ddp/db/dao/JdbiFormSectionBlock.sql.stg#L29

Reply:
Makes sense